mmdet/datasets/samplers/img_quota_sampler.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import math
+import random
+from typing import Iterator, Sized
+
+import numpy as np
+from mmengine.dataset import ClassBalancedDataset, ConcatDataset
+from mmengine.dist import get_dist_info
+from torch.utils.data import Sampler
+
+from mmdet.registry import DATA_SAMPLERS
+from ..base_video_dataset import BaseVideoDataset
+
+
+@DATA_SAMPLERS.register_module()
+class ImgQuotaSampler(Sampler):
+    """Sampler that gets fixed number of samples per epoch. It could be used in
+    both distributed and non-distributed environment.
+
+    Args:
+        dataset (Sized): Dataset used for sampling.
+        seed (int, optional): random seed used to shuffle the sampler. This
+            number should be identical across all processes in the distributed
+            group. Default: 0.
+    """
+
+    def __init__(self, dataset: Sized, seed: int = 0) -> None:
+        rank, world_size = get_dist_info()
+        self.rank = rank
+        self.world_size = world_size
+        self.epoch = 0
+        self.seed = seed if seed is not None else 0
```

Comment:
Forcing 0 doesn't quite make sense. We want to ensure full diversity when no seed is provided, so a different seed should be randomized when no seed is provided.  defaults to None is better